aisim/atoms.py

```diff
@@ -208,6 +212,9 @@ def create_random_ensemble_from_gaussian_distribution(pos_params, vel_params,
     ensemble : AtomicEnsemble
         Atomic ensemble containing the generated phase space vectors.
     """
+    if seed is not None:

```

Comment:
I think the seed function needs to be called after every single instance of generating random numbers.
Unfortunately, the seed just resets itself after generating a random

Reply:
After you set numpy.random.seed all random numbers generated after that are predictable:

```python
import numpy as np

np.random.seed(0)
rand1 = np.array([np.random.normal() for i in range(10)])

np.random.seed(0)
rand2 = np.array([np.random.normal() for i in range(10)])

rand1 == rand2
```

gives:

```python
array([ True,  True,  True,  True,  True,  True,  True,  True,  True, True])
```

Reply:
The setting is also global I think, so instead of passing `seed` you could also simply call `np.random.seed()`.

Reply:
@savowe : can you confirm?

Reply:
@bleykauf  yes. i did not think that also a second funciton call would give the same results but i tried
```
np.random.seed(0)
rand11 = np.array([np.random.normal() for i in range(10)])
rand12 = np.array([np.random.normal() for i in range(10)])

np.random.seed(0)
rand21 = np.array([np.random.normal() for i in range(10)])
rand22 = np.array([np.random.normal() for i in range(10)])

rand11 == rand21
rand12 == rand22
```
gives
```
>>> rand11 == rand21
array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True])
>>> rand12 == rand22
array([ True,  True,  True,  True,  True,  True,  True,  True,  True,
        True])
```
So I now think calling the functions once is good!.
If you put the random number generator inside a new function
```
def call_random(): return np.array([np.random.normal() for i in range(10)])

np.random.seed(0)
rand11 = call_random()
rand12 = call_random()

np.random.seed(0)
rand21 = call_random()
rand22 = call_random()

rand11 == rand21
rand12 == rand22
```
 it also works. So the scope goes at least into the function. I dont know whether it is now considered global